src/app/contacts/add-contact.component.html

```diff
@@ -0,0 +1,43 @@
+<form [formGroup]="contactForm" (ngSubmit)="onSave()">
+  <div class="form-group" [ngClass]="{'error': contactForm.controls.name.invalid && (contactForm.controls.name.dirty || submitted)}">
+    <label for="name">Name *</label>
+    <em *ngIf="(contactForm.controls.name.dirty || submitted) && contactForm.controls.name.errors?.required">Name is required</em>
```

Comment:
Gradnike, ki jih uporabljaš na formi, je fajn zapakirati v svoje komponente, da imaš potem povsod na enoten način poskrbljeno za error handling in se ti te dirty | submited | error kače ne vlečejo povsod :)

Reply:
Tole so res kače - sem se še danes zjutraj malo lovil, kot je razvidno iz commitov :). Najprej sem razmišljal, da bi za vsako polje izdelal ločen FormControl in bi lahko dostopal do polj direktno (npr. name.dirty in ne contactForm.controls.name.dirty). Tvoja rešitev je še boljša.